das/factory.go

```diff
@@ -148,3 +200,218 @@ func CreateBatchPosterDAS(
 
 	return daWriter, daReader, &lifecycleManager, nil
 }
+
+func CreateDAReaderWriterForStorage(
+	ctx context.Context,
+	config *DataAvailabilityConfig,
+	l1Reader *headerreader.HeaderReader,
+	seqInboxAddress *common.Address,
+) (DataAvailabilityServiceReader, DataAvailabilityServiceWriter, *LifecycleManager, error) {
+	if !config.Enable {
+		return nil, nil, nil, nil
+	}
+
+	// Check config requirements
+	if config.AggregatorConfig.Enable {
+		return nil, nil, nil, errors.New("--data-availability.rpc-aggregator")
+	}
+	if !config.LocalDBStorageConfig.Enable &&
+		!config.LocalFileStorageConfig.Enable &&
+		!config.S3StorageServiceConfig.Enable &&
+		!config.IpfsStorageServiceConfig.Enable {
+		return nil, nil, nil, errors.New("At least one of --data-availability.(local-db-storage|local-file-storage|s3-storage|ipfs-storage) must be enabled.")
+	}
```

Comment:
This input validation is duplicated from above. It's complex enough that it'd probably be worst making a helper function

Reply:
Addressed, see comment below.